grep-00XX-relicense-volk.md

```diff
@@ -0,0 +1,140 @@
+This approach should keep things simple.
+Most importantly, we do not introduce any new places to search for VOLK which would introduce confusion.
+
+#### how to
+We need to figure out how we collect all the contributors re-licensed code.
+This might be a file within the VOLK repo where everyone adds a signed-off commit that states they re-submit their code under the new license.
+Alternatively, we may collect all the contributors re-license statements in another place.
+Though, this might raise questions about long term documentation of this change.
+
+
+### The meantime
+
+It will probably take time until every contributor re-licenses their code.
+We want to be able to continue VOLK development while we are in this license transition period.
+A possible approach may be:
+
+- ask for informal support and start the formal process as soon as a majority of contributors confirmed their support.
+- Decide on a start and end date for this process
+    - This period should be between two releases.
+- Collect all the re-license agreements
+- Add a commit that reflects the license change in every necessary place
+- Do a first release under the new license.
+
+In the meantime, we use the following approach.
+First, we add a statement to the VOLK repository.
+```
+Starting at {DATE}, we can only accept contributions under the new license.
+```
+At that point, we probably notify every open PR about that new requirement.
+Then, we require
+```
+Copyright {YEAR} {CONTRIBUTOR}
+
+This file is part of VOLK
+
+SPDX-License-Identifier: LGPL-3.0-or-later
+```
+to be added to every modified file.
+cf. [SPDX identifier for LGPLv3+](https://spdx.org/licenses/LGPL-3.0-or-later.html).
+This entails a careful rework of the current license statements in all files.
+
+
+### The aftermath
+
+Assume most contributors re-licensed their VOLK contributions but some did not.
+How do we deal with that situation?
+
+- We could extend the transition period and hope contributors where just to busy to re-license their work yet.
+- Decide to remove contributions. This might imply that we lose features and change our API which either requires someone else to step up and re-implement this feature or definitely do a new major release that indicates API breakage.
+- Give up on the relicensing effort and revert to GPLv3+.
+
+A contributor who does not want or cannot re-license their work, cannot be forced or pushed to do so.
+
+We really hope to avoid this situation but need to be aware that we might face it anyways.
+Otherwise, we risk to stay in relicensing limbo.
+
+#### The first relicensed VOLK version
+
+We need to figure out the prefered way to announce the completed relicensing effort.
+We could consider the new license an API breaking change and release a new major version.
+Alternatively, we could continue the current major release series and just have another release.
```

Comment:
We should agree on what this is before completing. I think the breaking-change version update is the right way.

Reply:
Yes. Basically, at this point the license change is a breaking change. We will continue with a new major version.